api/block/schema.py

```diff
@@ -0,0 +1,28 @@
+import graphene
+from graphene import relay,Schema
+from graphene_sqlalchemy import (
+    SQLAlchemyObjectType, 
+    SQLAlchemyConnectionField
+)
+from api.block.models import Block as BlockModel
+
+class Block(SQLAlchemyObjectType):
+    class Meta:
+        model = BlockModel
+
+class Query(graphene.ObjectType):
+    node = relay.Node.Field()
+    floor = graphene.List(Block)
+    get_rooms_in_a_block = graphene.List(
+        lambda:Block,
+        block_id = graphene.Int()
+    )
+
+    def resolve_blocks(self,info):
+        query = Block.get_query(info)
+        return query.all()
+
+    def resolve_get_rooms_in_a_block(self,info,block_id):
+        query = Block.get_query(info)
+        result = query.filter(BlockModel.id == block_id)
+        return result
```

Comment:
The resolver method does not return what the name suggests **get rooms in a block**, actually what it returns is location based on the **block_id**. So I thick it should be **resolve_get_block**

Reply:
According to the flow on the front end, one needs to query rooms in a hierarchy. So the rooms_in_a_block will be from the block, the floor and finally rooms. Please refer to the screenshot.